cdap-api/src/main/java/io/cdap/cdap/api/feature/FeatureFlagsProvider.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package io.cdap.cdap.api.feature;
+
+/**
+ * Provides the ability to return feature flags
+ */
+public interface FeatureFlagsProvider {
+
+  /**
+   * This is an internal method. To find if a feature is enabled please use
+   * {@link io.cdap.cdap.features.Feature#isEnabled(FeatureFlagsProvider)}
+   *
+   * @return feature flag names to status ("true" or "false")
+   * if feature name is not defined throw IllegalArgumentException
+   */
+  boolean isFeatureEnabled(String name);
```

Comment:
As this is internal method, let's try to find a name that highlights it, e.g. `getFeatureConfigurationState`